### PR TITLE
Arch arm mpu move internal api

### PR DESCRIPTION
**Pull request by ioannisg**
A clean-up patch; moving an internal API header where it belongs (IMHO) :)

### CODE REVIEW

**Review by agansari — APPROVED** [review]
@ioannisg looks okay

File: arch/arm/core/cortex_m/mpu/arm_core_mpu.c [hunk 1]
```diff
@@ -9,7 +9,7 @@
 #include <kernel.h>
 #include <soc.h>
 
-#include <arch/arm/cortex_m/mpu/arm_core_mpu_dev.h>
+#include <arm_core_mpu_dev.h>
```
Comment (galak) [inline arch/arm/core/cortex_m/mpu/arm_core_mpu.c]: can this be "arm_core_mpu_dev.h" ?

Comment (ioannisg) [reply arch/arm/core/cortex_m/mpu/arm_core_mpu.c]: Yeap, changed this now.

File: tests/kernel/mem_protect/userspace/src/main.c [hunk 2]
```diff
@@ -23,7 +23,7 @@
 #endif
 
 #if defined(CONFIG_ARM)
-#include <arch/arm/cortex_m/mpu/arm_core_mpu_dev.h>
+extern void arm_core_mpu_disable(void);
```
Comment (galak) [inline tests/kernel/mem_protect/userspace/src/main.c]: So this made me look at the actual test, why do we have a disable_mmu_mpu() if we don't have an exposed API related to it?

**Review by galak — COMMENTED** [review]
Changes looks fine, but curious about why arm_core_mpu_disable has to get exported and what actually the test case is doing in disable_mmu_mpu().

Feels odd to me that we are "Testing" something that doesnt have an exposed API.

**Review by galak — DISMISSED** [review]
Can we remove arm_core_mpu_disable() from arm_core_mpu_dev.h since its effectively static and the only non-static use is the explicit extern in the testcase.

**Comment by ioannisg** [comment]
> 
> 
> Can we remove arm_core_mpu_disable() from arm_core_mpu_dev.h since its effectively static and the only non-static use is the explicit extern in the testcase.

Hm, I think yes - we used to invoke the enable/disable functions in arm_core_mpu.c, but perhaps not anymore.

**Comment by ioannisg** [comment]
> 
> 
> Can we remove arm_core_mpu_disable() from arm_core_mpu_dev.h since its effectively static and the only non-static use is the explicit extern in the testcase.

Done now, thanks.

File: tests/kernel/mem_protect/userspace/src/main.c (hunk 2 above)
Comment (ioannisg) [reply tests/kernel/mem_protect/userspace/src/main.c]: @andrewboie FYI

Comment (ioannisg) [reply tests/kernel/mem_protect/userspace/src/main.c]: @galak I get the feeling that this test is there to show that we can't disable the MPU from user mode, right @andrewboie ?

**Comment by galak** [comment]
LGTM, will wait to see if any discussion from @andrewboie on what the test is trying to do.

File: tests/kernel/mem_protect/userspace/src/main.c (hunk 2 above)
Comment (andrewboie) [reply tests/kernel/mem_protect/userspace/src/main.c]: I wouldn't object if we deleted that test. There's some other dubious cases in there as well.

**Review by tyrantlizard77 — APPROVED** [review]

File: tests/kernel/mem_protect/userspace/src/main.c (hunk 2 above)
Comment (ioannisg) [reply tests/kernel/mem_protect/userspace/src/main.c]: I like that it "tests" that user-mode can't play with the mpu :)

Comment (andrewboie) [reply tests/kernel/mem_protect/userspace/src/main.c]: Are we trying to prove whether the CPU works or the software?
Proving that the CPU works is a waste of time IMO

Comment (ioannisg) [reply tests/kernel/mem_protect/userspace/src/main.c]: Yeap, that's in fact, correct

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**